src/sincosint.jl

```diff
@@ -0,0 +1,258 @@
+const pidiv2 = π/2
+
+import Base.Math: @horner
+
+# Compute the sine integral: ∫_0^x sin(t)/t dt,
+# and the cosine integral: γ + log x + ∫_0^x (cos(t)-1)/t dt,
+# using the rational approximants tabulated in:
+#     A.J. MacLeod, "Rational approximations, software and test methods for
+#     sine and cosine integrals," Numer. Algor. 12, pp. 259--272 (1996).
+#         http://dx.doi.org/10.1007/BF02142806
+#         https://link.springer.com/article/10.1007/BF02142806
+#
+# Note: the second zero of Ci(x) has a typo that is fixed:
+#
+# r1 = 3.38418 04228 51186 42639 78511 46402 in the article, but is in fact:
+#
+# r1 = 3.38418 04225 51186 42639 78511 46402.
+#
+
+function sinint(x::Float64)
+    t = x*x
+    if t ≤ 36.0
+        return x * @horner(t, 1.00000000000000000000E0,
+                             -0.44663998931312457298E-1,
+                              0.11209146443112369449E-2,
+                             -0.13276124407928422367E-4,
+                              0.85118014179823463879E-7,
+                             -0.29989314303147656479E-9,
+                              0.55401971660186204711E-12,
+                             -0.42406353433133212926E-15) /
+                   @horner(t, 1.00000000000000000000E0,
+                              0.10891556624243098264E-1,
+                              0.59334456769186835896E-4,
+                              0.21231112954641805908E-6,
+                              0.54747121846510390750E-9,
+                              0.10378561511331814674E-11,
+                              0.13754880327250272679E-14,
+                              0.10223981202236205703E-17)
+    elseif t ≤ 144.0
+        invt = inv(t)
```

Comment:
I agree, 1 ulp is too demanding for complex division.

I wonder if Julia's division of a real by a complex should be implemented as follows:
```julia
julia> import Base: /

julia> /(x::Real, y::Complex) = inv(y/x)
WARNING: Method definition /(Real, Base.Complex{T} where T<:Real) in module Base at complex.jl:278 overwritten in module Main at REPL[7]:1.
/ (generic function with 73 methods)

julia> x = 49.0
49.0

julia> y = 49.0+0.0im
49.0 + 0.0im

julia> x/y
1.0 - 0.0im

```
This achieves two things: it delegates conversion & promotion to `/(x::Complex, y::Real)`, and except possibly for corner cases it replicates `/(x::Real,y::Real)`, which is intuitively satisfying.